brave/brave-core
# [ads] Allow non-Rewards Android users to clear Brave Ads data from `Delete browsing data`

The PR adds `Clear Brave Ads data...` and `Reset Brave Rewards data...` links to the `Delete browsing data` dialog in settings. The Desktop has similar links for `Delete browsing data` settings.

`Clear Brave Ads data...` link is shown for non-Rewards users. Clicking the link deletes Brave Ads data from ads_service directory in Profile and restarts Brave Ads service:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/c1e4e7db-98de-48b4-89d8-50fafbb7c561">


`Reset Brave Rewards data...` link is shown for Rewards users. Clicking the link opens a new tab with `brave://rewards/#reset` URL:

<img width="300" alt="image" src="https://github.com/user-attachments/assets/062e8298-d04c-4208-a390-527d254f7a66">


Resolves https://github.com/brave/brave-browser/issues/40290



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Test case 1
* Fresh install
* Make sure that Brave Ads service is started.
     EXPECTATION: There should be log line:
    `Successfully initialized ads`
* Clear the Rewards log
* Open `Settings` -> `History` -> `Delete browsing data`
* Tap `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
     EXPECTATION: There should be log line:
    `Initializing ads`
* Check Brave Ads database was deleted and recreated:
     EXPECTATION: There should be log line:
    `Created database for schema version `
* Check that `Brave Ads` client state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Client state does not exist, creating default state`
* Check that `Brave Ads` confirmation state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Confirmation state does not exist, creating default state`

### Test case 2
* Fresh install
* Join Brave News
* Make sure that Brave Ads service is started.
     EXPECTATION: There should be log line:
    `Successfully initialized ads`
* Clear the Rewards log
* Open `Settings` -> `Brave Shields & privacy` -> `Delete browsing data` -> `Advanced` tab
* Tap `Clear Brave Ads data...` link
* Make sure that Brave Ads service is restarted
     EXPECTATION: There should be log line:
    `Initializing ads`
* Check Brave Ads database was deleted and recreated:
     EXPECTATION: There should be log line:
    `Created database for schema version `
* Check that `Brave Ads` client state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Client state does not exist, creating default state`
* Check that `Brave Ads` confirmation state file was deleted and recreated:
     EXPECTATION: There should be log line:
    `Confirmation state does not exist, creating default state`

### Test case 3
* Fresh install
* Join Brave Rewards
* Open `Settings` -> `History` -> `Delete browsing data`
* Make sure `Reset Brave Rewards data...` link is shown
* Open `Settings` -> `Brave Shields & privacy` -> `Delete browsing data` -> `Advanced` tab
* Make sure `Reset Brave Rewards data...` link is shown
